### ISSUE TITLE
Deal with ::: nonsense

### ISSUE THREAD

**Issue by dill** [DistanceDevelopment/dsm#17]
At the moment there is an issue when submitting `dsm` to CRAN that the `dsm.var.prop` method uses `:::` to import functions from `mrds` that are not in its `NAMESPACE`. One way to get around this would be to export these functions in `mrds` and just give them the keyword `internal` in their documentation.

Is this a reasonable solution?

**Comment by LHMarshall** [comment]
Which functions from dsm are required in this way from mrds?

**Comment by dill** [comment]
From `R CMD CHECK dsm_2.2.11.tar.gz`:

```
* checking dependencies in R code ... NOTE
Unexported objects imported by ':::' calls:
  ‘mrds:::assign.par’ ‘mrds:::create.ddfobj’ ‘mrds:::detfct’ ‘mrds:::process.data’
  See the note in ?`:::` about the use of this operator.
```

**Comment by dill** [comment]
To add a little more to this...

Used by `dsm.var.prop`: `mrds:::assign.par`
Used by `generate.ds.uncertainty`: `mrds:::detfct` `mrds:::create.ddfobj` `mrds:::process.data`

So there are two separate issues here -- one with variance propagation and one with the detection function uncertainty when doing bootstraps.

**Comment by dill** [comment]
`assign.par` is now exported from `mrds` (as of [7c1c5ce](https://github.com/DistanceDevelopment/mrds/commit/7c1c5ceeca00e869349c5dd89e43708afaa75f80)), code fixed in `dsm` as of 21b1720.

**Comment by dill** [comment]
b54c213 removes the dependency on `mrds:::process.data` from `generate.ds.uncertainty`.

**Comment by dill** [comment]
Fixed in 2103a63, taking into account `mrds` changes in [e4e8d50](https://github.com/DistanceDevelopment/mrds/commit/e4e8d50ed6f08e467417b2156ae113ef6b5b6a49).
